docs/migrating-from-0.5.md

```diff
@@ -2,7 +2,127 @@
 
 Genkit 0.9 introduces a number of breaking changes alongside feature enhancements that improve overall functionality. If you have been developing applications with Genkit 0.5, you will need to update your application code when you upgrade to the latest version. This guide outlines the most significant changes and offers steps to migrate your existing applications smoothly.
 
-## 1. CLI Changes
+## Quickstart guide
+
+The following steps will get you up and running on Genkit 0.9 quickly. Read more information about these changes in the detailed [Changelog](#changelog) below.
+
+### 1. Install the new CLI
+
+* Uninstall the old CLI
+
+```posix-terminal
+npm uninstall -g genkit
+```
+
+* Install the new CLI
+
+```posix-terminal
+# Install globally
+npm i -g genkit-cli
+
+# Or install as a dev dependency
+npm i -D genkit-cli
+```
+
+### 2. Update your dependencies
+
+* Remove individual Genkit core packages
+
+```posix-terminal
+npm uninstall @genkit-ai/ai @genkit-ai/core @genkit-ai/dotprompt @genkit-ai/flow
+```
+
+* Install the new consolidated `genkit` package
+
+```posix-terminal
+npm i --save genkit
+```
+
+* Upgrade all plugin versions (example below)
+
+```
+npm upgrade @genkit-ai/firebase
+```
+
+### 3. Change your imports
+
+* Remove imports from individual Genkit core packages
+
+```js
+import { … } from '@genkit-ai/ai';
+import { … } from '@genkit-ai/core';
+import { … } from '@genkit-ai/flow';
+```
+
+* Remove zod imports
+
+```js
+import * as z from 'zod';
+```
+  
+* Import `genkit` and `zod` from `genkit`
+
+```js
+import { z, genkit } from 'genkit';
+```
+
+### 4. Update your code
+
+#### Configuration is now much simpler 
+
+Configuration for Genkit is now done per instance. Telemetry and logging is configured globally and separately from the Genkit instance. 
+
+See the [Monitoring and Logging](./monitoring.md) documentation for more details on how to configure telemetry and logging.
+
+See the [Get Started](./get-started.md) documentation for more details on how to configure a Genkit instance.
+
+```js
+import { genkit } from 'genkit';
+import { logger } from 'genkit/logging';
+import { enableFirebaseTelemetry } from '@genkit-ai/firebase';
+
+logger.setLogLevel('debug');
+enableFirebaseTelemetry({...});
+
+const ai = genkit({ plugins: [...]});
+```
+
+#### Genkit primitives are now registered to a specific `genkit` instance
```

Comment:
Again, imperative

```Replace defineFlow, definePropmpt, defineTools with their instance-based versions```
 Action definitions (flows, tools, retrievers, indexers, etc.) are now only defined per instance instead of Globally. Anywhere you previously had a `defineFlow` replace it with a `ai.defineFlow` where `ai` is your `Genkit` instance.

Reply:
I instead said "Migrate Genkit primitives to be called from the `genkit` instance" - wdyt?

Reply:
I think we ended up calling them `actions` instead of `primitives`

Reply:
Done.